src/Orleans.Transactions/State/TransactionQueue.cs

```diff
@@ -225,6 +228,8 @@ public void NotifyOfAbort(TransactionRecord<TState> entry, TransactionalStatus s
                         if (logger.IsEnabled(LogLevel.Trace))
                             logger.Trace($"aborting RemoteCommitEntry {entry.Timestamp:o} status={status}");
 
+                        entry.ConfirmationResponsePromise?.TrySetException(new OrleansException($"Confirm failed: Status {status}"));
```

Comment:
Code is unreachable, should never happen: cannot receive an abort if the transaction has committed on the TM.

Reply:
NotifyOfAbort is called during any storage error.  If a storage error occurs when a confirmation is being processed, the promise will never be resolved, and the TM will timeout.  Sending this exception allows the TM to recover faster.  It's not strictly necessary, but recovery is delayed greatly when we don't resolve pending operations and rely on orleans timeouts.

Reply:
ok, you are right - I mistook the "NotifyOfAbort" method for the code that handles the "Abort" message sent by the TA. But that one is called Abort. 

Is there some ambiguity with the "NotifyOf" prefix? Here it seems to mean "notify other participants of", but in other cases it means "notify this participant of".